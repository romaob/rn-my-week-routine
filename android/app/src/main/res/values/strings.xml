<resources>
    <string name="app_name">MyWeekRoutine</string>
</resources>
