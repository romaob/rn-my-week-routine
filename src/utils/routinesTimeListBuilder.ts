/*
    Given a routine list, time slots list, and current day index.
    It should build and return the objcets for being rendered on the screen.

    It should follow the following logic:
    - Filter events array, returning only those that are for the current day.
    - Go over every time slot
        - check if there is a routine for that time slot.
            - if there is, check if there is a blockRow for that time slot and select it, if not, create it.
            - add a blockColumn to the blockRow.
                - check if is there empty space before the routine.
                    - if there is, add a emptyRoutineBlocks to the columnBlock until filling the empty space.
                - add a routineBlock to the columnBlock.
                    - Fill the routineBlock with rows representing all time slots that the routine takes.
                - Check if there is empty space after the routine.
                    - if there is, add a emptyRoutineBlocks to the columnBlock until filling the empty space.
                - Remove the routine from the list.
        - Repeats until there are no more routines for the time slot
*/