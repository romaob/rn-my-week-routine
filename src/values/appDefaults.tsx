export enum APP_KEYS {
  LANGUAGE = '@myweekroutine.language',
  ALERTS_ENABLED = '@myweekroutine.alertsEnabled',
  ONBOARDING = '@myweekroutine.onboarding',
  EVENTS = '@myweekroutine.events',
}

export interface Event {
  id: string;
  name: string;
  indexes: number[];
  description: string;
  startAt: string;
  endAt: string;
  alertEnabled: boolean;
  alertSent: boolean;
  alertConfirmed: boolean;
  added: string;
  updated: string;
}
