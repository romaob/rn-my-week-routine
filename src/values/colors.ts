// App color palette
export const colors = {
  light: {
    primary: '#3F51B5',
    secondary: '#536DFE',
    accent: '#FF4081',
    danger: '#f44336',
    warning: '#ff9800',
    success: '#4caf50',
    info: '#2196f3',
    background: '#ffffff',
    cardBackground: '#ffffff',
    disabled: '#bdbdbd',
    text: '#000000',
    textContrast: '#ffffff',
    textSecondary: '#757575',
    textDisabled: '#bdbdbd',
  },
};
