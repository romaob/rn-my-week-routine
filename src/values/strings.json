{
    "app_name": {
        "en": "My Week Routine",
        "es": "Mi Rutina Semanal",
        "pt": "Minha Rotina Semanal"
    },
    "screen_routine": {
        "en": "Routine",
        "es": "Rutina",
        "pt": "Rotina"
    },
    "screen_events": {
        "en": "My Events",
        "es": "Mis Eventos",
        "pt": "Meus Eventos"
    },
    "screen_settings": {
        "en": "Settings",
        "es": "Ajustes",
        "pt": "Configurações"
    },
    "screen_about": {
        "en": "About",
        "es": "Acerca de",
        "pt": "Sobre"
    },
    "lang_en": {
        "en": "English",
        "es": "English",
        "pt": "English"
    },
    "lang_es": {
        "en": "Español",
        "es": "Español",
        "pt": "Español"
    },
    "lang_pt": {
        "en": "Português",
        "es": "Português",
        "pt": "Português"
    },
    "test": {
        "en": "Testing the app strings",
        "es": "Probando las cadenas de la aplicación",
        "pt": "Testando as strings do app"
    },
    "week_day_mon": {
        "en": "Monday",
        "es": "Lunes",
        "pt": "Segunda"
    },
    "week_day_tue": {
        "en": "Tuesday",
        "es": "Martes",
        "pt": "Terça"
    },
    "week_day_wed": {
        "en": "Wednesday",
        "es": "Miércoles",
        "pt": "Quarta"
    },
    "week_day_thu": {
        "en": "Thursday",
        "es": "Jueves",
        "pt": "Quinta"
    },
    "week_day_fri": {
        "en": "Friday",
        "es": "Viernes",
        "pt": "Sexta"
    },
    "week_day_sat": {
        "en": "Saturday",
        "es": "Sábado",
        "pt": "Sábado"
    },
    "week_day_sun": {
        "en": "Sunday",
        "es": "Domingo",
        "pt": "Domingo"
    },
    "save": {
        "en": "Save",
        "es": "Guardar",
        "pt": "Salvar"
    },
    "delete": {
        "en": "Delete",
        "es": "Eliminar",
        "pt": "Excluir"
    },
    "cancel": {
        "en": "Cancel",
        "es": "Cancelar",
        "pt": "Cancelar"
    },
    "ok": {
        "en": "OK",
        "es": "OK",
        "pt": "OK"
    },
    "routine_new": {
        "en": "New Routine",
        "es": "Nueva Rutina",
        "pt": "Nova Rotina"
    },
    "routine_name": {
        "en": "Routine Name",
        "es": "Nombre de la Rutina",
        "pt": "Nome da Rotina"
    },
    "routine_description": {
        "en": "Routine Description",
        "es": "Descripción de la Rutina",
        "pt": "Descrição da Rotina"
    },
    "routine_days_select": {
        "en": "On which days do you want to do this routine?",
        "es": "¿En qué días quieres hacer esta rutina?",
        "pt": "Em quais dias você quer fazer esta rotina?"
    },
    "routine_start_time": {
        "en": "When it will start?",
        "es": "¿Cuándo va a empezar?",
        "pt": "Quando vai começar?"
    },
    "routine_end_time": {
        "en": "When it will end?",
        "es": "¿Cuándo va a terminar?",
        "pt": "Quando vai terminar?"
    },
    "routine_delete_title": {
        "en": "Delete Routine",
        "es": "Eliminar Rutina",
        "pt": "Excluir Rotina"
    },
    "routine_delete_message": {
        "en": "Are you sure you want to delete this routine?",
        "es": "¿Estás seguro de que quieres eliminar esta rutina?",
        "pt": "Tem certeza que deseja excluir esta rotina?"
    },
    "routine_name_error": {
        "en": "Routine name is required",
        "es": "El nombre de la rutina es obligatorio",
        "pt": "O nome da rotina é obrigatório"
    },
    "routine_start_at": {
        "en": "Start at",
        "es": "Empieza a las",
        "pt": "Começa às"
    },
    "routine_end_at": {
        "en": "End at",
        "es": "Termina a las",
        "pt": "Termina às"
    },
    "screen_settings_language": {
        "en": "Language",
        "es": "Idioma",
        "pt": "Idioma"
    },
    "screen_settings_about": {
        "en": "About",
        "es": "Acerca de",
        "pt": "Sobre"
    },
    "screen_settings_clear_events": {
        "en": "Clear Events",
        "es": "Limpiar Eventos",
        "pt": "Limpar Eventos"
    }
}