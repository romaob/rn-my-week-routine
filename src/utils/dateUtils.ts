import strings from '../values/strings.json';

export const WEEK_DAYS = [
  'week_day_sun',
  'week_day_mon',
  'week_day_tue',
  'week_day_wed',
  'week_day_thu',
  'week_day_fri',
  'week_day_sat',
];

export function getWeekDays(language: string = 'en'): string[] {
  return WEEK_DAYS.map(day => strings[day][language]);
}

export function getTimeStringFromDate(date: Date): string {
  return `${date.getHours()}:${date.getMinutes()}`;
}

export function getDateIncreasedByMinutes(date: Date, minutes: number): Date {
  return new Date(date.getTime() + minutes * 60000);
}

export function getDayTimesForMinutes(minutes: number): Date[] {
  const result = [];
  const date = new Date(new Date().setHours(0, 0, 0, 0));
  for (let i = 0; i < (24 * 60) / minutes; i++) {
    result.push(new Date(date.getTime() + i * minutes * 60000));
  }
  return result;
}

export function getSlotIndexOfDate(date: Date, minutes: number): number {
  return (
    new Date(date).getHours() * 2 +
    (new Date(date).getMinutes() >= minutes ? 1 : 0)
  );
}
