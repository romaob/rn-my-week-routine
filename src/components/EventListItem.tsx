import React from 'react';
import {getTimeStringFromDate, getWeekDays} from '../utils/dateUtils';
import {StyleSheet, View} from 'react-native';
import Label, {FontSize} from './Label';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {Event} from '../values/appDefaults';
import {colors} from '../values/colors';
import {sizes} from '../values/sizes';
import {useString} from '../context/useStringContext';

function WeekDaysListItem({
  day,
  selected,
}: {
  day: string;
  selected: boolean;
}): JSX.Element {
  return (
    <View
      style={{
        ...styles.weekDaysListItem,
        ...(selected ? styles.weekDaysListItemSelected : {}),
      }}
      testID="eventListItemWeekDay">
      <Label text={day.charAt(0).toUpperCase()} size={FontSize.SMALL} />
    </View>
  );
}

export default function EventListItem({
  item,
  onLongPress,
}: {
  item: Event;
  onLongPress?: (event: Event) => void;
}): JSX.Element {
  const {language, getString} = useString();
  const weekDays = getWeekDays(language);

  return (
    <TouchableOpacity
      testID="eventListItem"
      style={styles.listItem}
      onLongPress={() => onLongPress && onLongPress(item)}>
      <Label
        text={item.name}
        style={styles.title}
        color={colors.light.accent}
        testID="eventListItemName"
      />
      <Label
        text={item.description}
        size={FontSize.MEDIUM}
        color={colors.light.textSecondary}
        testID="eventListItemDescription"
      />
      <View style={styles.listItemTimes}>
        <Label
          text={
            getString('routine_start_at') +
            ' ' +
            getTimeStringFromDate(new Date(item.startAt))
          }
          size={FontSize.SMALL}
          color={colors.light.textSecondary}
          style={styles.listItemTimesText}
          testID="eventListItemStart"
        />
        <Label
          text={
            getString('routine_end_at') +
            ' ' +
            getTimeStringFromDate(new Date(item.endAt))
          }
          size={FontSize.SMALL}
          color={colors.light.textSecondary}
          style={styles.listItemTimesText}
          testID="eventListItemEnd"
        />
      </View>
      <View style={styles.listItemFooter}>
        {weekDays.map((day, index) => (
          <WeekDaysListItem
            key={index}
            day={day}
            selected={item.indexes.includes(index)}
          />
        ))}
      </View>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.light.background,
    paddingVertical: sizes.padding.sm,
  },
  listItem: {
    display: 'flex',
    margin: sizes.margin.sm,
    padding: sizes.padding.sm,
    minHeight: 40,
    backgroundColor: colors.light.cardBackground,
    gap: sizes.margin.sm,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: colors.light.grey + '70',
  },
  title: {
    fontWeight: 'bold',
    fontSize: 18,
  },
  listItemTimes: {
    display: 'flex',
    flexDirection: 'row',
  },
  listItemTimesText: {
    fontWeight: 'bold',
    flex: 1,
  },
  listItemFooter: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-end',
  },
  weekDaysListItem: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    margin: sizes.margin.sm,
    borderRadius: 999,
    padding: 3,
    width: 25,
    height: 25,
    backgroundColor: colors.light.grey + '30',
    borderWidth: 1,
    borderColor: colors.light.grey + '50',
  },
  weekDaysListItemSelected: {
    backgroundColor: colors.light.accent + '70',
  },
});
