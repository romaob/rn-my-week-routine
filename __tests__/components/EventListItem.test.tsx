import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import EventListItem from '../../src/components/EventListItem';
import {Event} from '../../src/values/appDefaults';
import {getTimeStringFromDate, getWeekDays} from '../../src/utils/dateUtils';
import {colors} from '../../src/values/colors';

import strings from '../../src/values/strings.json';
import StringsProvider from '../../src/context/useStringContext';

const testEvent: Event = {
  id: '1234',
  name: 'Test Name',
  description: 'Test Description',
  startAt: '2023-09-13T12:00:00.000',
  endAt: '2023-09-13T12:30:00.000',
  indexes: [2, 3, 4],
  alertEnabled: false,
  alertSent: false,
  alertConfirmed: false,
  added: new Date().toISOString(),
  updated: new Date().toISOString(),
};

describe('EventListItem render tests', () => {
  it('should render a Event item correctly', () => {
    const result = render(
      <StringsProvider>
        <EventListItem item={testEvent} />
      </StringsProvider>,
    );
    expect(result).toBeDefined();
    expect(result).toMatchSnapshot();
    expect(result.getByTestId('eventListItem')).toBeDefined();
    //Testing event name
    expect(result.getByTestId('eventListItemName')).toBeDefined();
    expect(result.getByText(testEvent.name)).toBeDefined();
    expect(
      result
        .getByTestId('eventListItemName')
        .props.children.some((child: any) => child === testEvent.name),
    ).toBeTruthy();
    expect(result.getByTestId('eventListItemName').props.style.fontWeight).toBe(
      'bold',
    );
    expect(result.getByTestId('eventListItemName').props.style.color).toBe(
      colors.light.accent,
    );

    //Testing event description
    expect(result.getByTestId('eventListItemDescription')).toBeDefined();
    expect(result.getByText(testEvent.description)).toBeDefined();
    expect(
      result
        .getByTestId('eventListItemDescription')
        .props.children.some((child: any) => child === testEvent.description),
    ).toBeTruthy();
    //Testing event startAt
    expect(result.getByTestId('eventListItemStart')).toBeDefined();
    expect(
      result.getByText(
        strings.routine_start_at.en +
          ' ' +
          getTimeStringFromDate(new Date(testEvent.startAt)),
      ),
    ).toBeDefined();
    expect(
      result
        .getByTestId('eventListItemStart')
        .props.children.some(
          (child: any) =>
            child ===
            strings.routine_start_at.en +
              ' ' +
              getTimeStringFromDate(new Date(testEvent.startAt)),
        ),
    ).toBeTruthy();
    expect(
      result.getByTestId('eventListItemStart').props.style.fontWeight,
    ).toBe('bold');
    //Testing event endAt
    expect(result.getByTestId('eventListItemEnd')).toBeDefined();
    expect(
      result.getByText(
        strings.routine_end_at.en +
          ' ' +
          getTimeStringFromDate(new Date(testEvent.endAt)),
      ),
    ).toBeDefined();
    expect(
      result
        .getByTestId('eventListItemEnd')
        .props.children.some(
          (child: any) =>
            child ===
            strings.routine_end_at.en +
              ' ' +
              getTimeStringFromDate(new Date(testEvent.endAt)),
        ),
    ).toBeTruthy();
    expect(result.getByTestId('eventListItemEnd').props.style.fontWeight).toBe(
      'bold',
    );
    //Testing event weekDays
    const weekDaysItems = result.getAllByTestId('eventListItemWeekDay');
    const weekDaysTest = getWeekDays('en');
    expect(weekDaysItems).toBeDefined();
    expect(weekDaysItems.length).toBe(7);
    for (let i = 0; i < 7; i++) {
      expect(weekDaysItems[i]).toBeDefined();
      //Testing the text content
      const labelChild = weekDaysItems[i].props.children;
      expect(labelChild.props.text).toBe(
        weekDaysTest[i].charAt(0).toUpperCase(),
      );
      //Testing the style
      expect(weekDaysItems[i].props.style.backgroundColor).toBe(
        testEvent.indexes.includes(i)
          ? colors.light.accent + 70
          : colors.light.grey + 30,
      );
    }
  });
});

describe('EventListItem action tests', () => {
  it('should call the provided funcion on the long press', () => {
    const onLongPress = jest.fn();
    const result = render(
      <StringsProvider>
        <EventListItem item={testEvent} onLongPress={onLongPress} />
      </StringsProvider>,
    );

    expect(result).toBeDefined();
    expect(result).toMatchSnapshot();
    expect(result.getByTestId('eventListItem')).toBeDefined();
    //Testing the long press
    expect(onLongPress).not.toHaveBeenCalled();
    fireEvent(result.getByTestId('eventListItem'), 'longPress');
    expect(onLongPress).toHaveBeenCalled();
  });
});
