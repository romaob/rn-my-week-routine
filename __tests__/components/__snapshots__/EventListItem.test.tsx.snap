// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventListItem action tests should call the provided funcion on the long press 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={2}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  testID="eventListItem"
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "backgroundColor": "#ffffff",
        "borderColor": "#9e9e9e70",
        "borderRadius": 10,
        "borderWidth": 1,
        "display": "flex",
        "gap": 8,
        "margin": 8,
        "minHeight": 40,
        "opacity": 1,
        "padding": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "#d81e8b",
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
      testID="eventListItemName"
    >
      Test Name
    </Text>
    <Text
      style={
        {
          "color": "#757575",
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 14,
        }
      }
      testID="eventListItemDescription"
    >
      Test Description
    </Text>
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "color": "#757575",
            "display": "flex",
            "flex": 1,
            "flexDirection": "row",
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
        testID="eventListItemStart"
      >
        Start at 12:00 PM
      </Text>
      <Text
        style={
          {
            "color": "#757575",
            "display": "flex",
            "flex": 1,
            "flexDirection": "row",
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
        testID="eventListItemEnd"
      >
        End at 12:30 PM
      </Text>
    </View>
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#9e9e9e30",
            "borderColor": "#9e9e9e50",
            "borderRadius": 999,
            "borderWidth": 1,
            "display": "flex",
            "height": 25,
            "justifyContent": "center",
            "margin": 8,
            "padding": 3,
            "width": 25,
          }
        }
        testID="eventListItemWeekDay"
      >
        <Text
          style={
            {
              "color": "#000000",
              "display": "flex",
              "flexDirection": "row",
              "fontSize": 12,
            }
          }
          testID="label"
        >
          S
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#9e9e9e30",
            "borderColor": "#9e9e9e50",
            "borderRadius": 999,
            "borderWidth": 1,
            "display": "flex",
            "height": 25,
            "justifyContent": "center",
            "margin": 8,
            "padding": 3,
            "width": 25,
          }
        }
        testID="eventListItemWeekDay"
      >
        <Text
          style={
            {
              "color": "#000000",
              "display": "flex",
              "flexDirection": "row",
              "fontSize": 12,
            }
          }
          testID="label"
        >
          M
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#d81e8b70",
            "borderColor": "#9e9e9e50",
            "borderRadius": 999,
            "borderWidth": 1,
            "display": "flex",
            "height": 25,
            "justifyContent": "center",
            "margin": 8,
            "padding": 3,
            "width": 25,
          }
        }
        testID="eventListItemWeekDay"
      >
        <Text
          style={
            {
              "color": "#000000",
              "display": "flex",
              "flexDirection": "row",
              "fontSize": 12,
            }
          }
          testID="label"
        >
          T
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#d81e8b70",
            "borderColor": "#9e9e9e50",
            "borderRadius": 999,
            "borderWidth": 1,
            "display": "flex",
            "height": 25,
            "justifyContent": "center",
            "margin": 8,
            "padding": 3,
            "width": 25,
          }
        }
        testID="eventListItemWeekDay"
      >
        <Text
          style={
            {
              "color": "#000000",
              "display": "flex",
              "flexDirection": "row",
              "fontSize": 12,
            }
          }
          testID="label"
        >
          W
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#d81e8b70",
            "borderColor": "#9e9e9e50",
            "borderRadius": 999,
            "borderWidth": 1,
            "display": "flex",
            "height": 25,
            "justifyContent": "center",
            "margin": 8,
            "padding": 3,
            "width": 25,
          }
        }
        testID="eventListItemWeekDay"
      >
        <Text
          style={
            {
              "color": "#000000",
              "display": "flex",
              "flexDirection": "row",
              "fontSize": 12,
            }
          }
          testID="label"
        >
          T
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#9e9e9e30",
            "borderColor": "#9e9e9e50",
            "borderRadius": 999,
            "borderWidth": 1,
            "display": "flex",
            "height": 25,
            "justifyContent": "center",
            "margin": 8,
            "padding": 3,
            "width": 25,
          }
        }
        testID="eventListItemWeekDay"
      >
        <Text
          style={
            {
              "color": "#000000",
              "display": "flex",
              "flexDirection": "row",
              "fontSize": 12,
            }
          }
          testID="label"
        >
          F
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#9e9e9e30",
            "borderColor": "#9e9e9e50",
            "borderRadius": 999,
            "borderWidth": 1,
            "display": "flex",
            "height": 25,
            "justifyContent": "center",
            "margin": 8,
            "padding": 3,
            "width": 25,
          }
        }
        testID="eventListItemWeekDay"
      >
        <Text
          style={
            {
              "color": "#000000",
              "display": "flex",
              "flexDirection": "row",
              "fontSize": 12,
            }
          }
          testID="label"
        >
          S
        </Text>
      </View>
    </View>
  </View>
</RNGestureHandlerButton>
`;

exports[`EventListItem render tests should render a Event item correctly 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  testID="eventListItem"
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "backgroundColor": "#ffffff",
        "borderColor": "#9e9e9e70",
        "borderRadius": 10,
        "borderWidth": 1,
        "display": "flex",
        "gap": 8,
        "margin": 8,
        "minHeight": 40,
        "opacity": 1,
        "padding": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "#d81e8b",
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
      testID="eventListItemName"
    >
      Test Name
    </Text>
    <Text
      style={
        {
          "color": "#757575",
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 14,
        }
      }
      testID="eventListItemDescription"
    >
      Test Description
    </Text>
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "color": "#757575",
            "display": "flex",
            "flex": 1,
            "flexDirection": "row",
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
        testID="eventListItemStart"
      >
        Start at 12:00 PM
      </Text>
      <Text
        style={
          {
            "color": "#757575",
            "display": "flex",
            "flex": 1,
            "flexDirection": "row",
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
        testID="eventListItemEnd"
      >
        End at 12:30 PM
      </Text>
    </View>
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#9e9e9e30",
            "borderColor": "#9e9e9e50",
            "borderRadius": 999,
            "borderWidth": 1,
            "display": "flex",
            "height": 25,
            "justifyContent": "center",
            "margin": 8,
            "padding": 3,
            "width": 25,
          }
        }
        testID="eventListItemWeekDay"
      >
        <Text
          style={
            {
              "color": "#000000",
              "display": "flex",
              "flexDirection": "row",
              "fontSize": 12,
            }
          }
          testID="label"
        >
          S
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#9e9e9e30",
            "borderColor": "#9e9e9e50",
            "borderRadius": 999,
            "borderWidth": 1,
            "display": "flex",
            "height": 25,
            "justifyContent": "center",
            "margin": 8,
            "padding": 3,
            "width": 25,
          }
        }
        testID="eventListItemWeekDay"
      >
        <Text
          style={
            {
              "color": "#000000",
              "display": "flex",
              "flexDirection": "row",
              "fontSize": 12,
            }
          }
          testID="label"
        >
          M
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#d81e8b70",
            "borderColor": "#9e9e9e50",
            "borderRadius": 999,
            "borderWidth": 1,
            "display": "flex",
            "height": 25,
            "justifyContent": "center",
            "margin": 8,
            "padding": 3,
            "width": 25,
          }
        }
        testID="eventListItemWeekDay"
      >
        <Text
          style={
            {
              "color": "#000000",
              "display": "flex",
              "flexDirection": "row",
              "fontSize": 12,
            }
          }
          testID="label"
        >
          T
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#d81e8b70",
            "borderColor": "#9e9e9e50",
            "borderRadius": 999,
            "borderWidth": 1,
            "display": "flex",
            "height": 25,
            "justifyContent": "center",
            "margin": 8,
            "padding": 3,
            "width": 25,
          }
        }
        testID="eventListItemWeekDay"
      >
        <Text
          style={
            {
              "color": "#000000",
              "display": "flex",
              "flexDirection": "row",
              "fontSize": 12,
            }
          }
          testID="label"
        >
          W
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#d81e8b70",
            "borderColor": "#9e9e9e50",
            "borderRadius": 999,
            "borderWidth": 1,
            "display": "flex",
            "height": 25,
            "justifyContent": "center",
            "margin": 8,
            "padding": 3,
            "width": 25,
          }
        }
        testID="eventListItemWeekDay"
      >
        <Text
          style={
            {
              "color": "#000000",
              "display": "flex",
              "flexDirection": "row",
              "fontSize": 12,
            }
          }
          testID="label"
        >
          T
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#9e9e9e30",
            "borderColor": "#9e9e9e50",
            "borderRadius": 999,
            "borderWidth": 1,
            "display": "flex",
            "height": 25,
            "justifyContent": "center",
            "margin": 8,
            "padding": 3,
            "width": 25,
          }
        }
        testID="eventListItemWeekDay"
      >
        <Text
          style={
            {
              "color": "#000000",
              "display": "flex",
              "flexDirection": "row",
              "fontSize": 12,
            }
          }
          testID="label"
        >
          F
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#9e9e9e30",
            "borderColor": "#9e9e9e50",
            "borderRadius": 999,
            "borderWidth": 1,
            "display": "flex",
            "height": 25,
            "justifyContent": "center",
            "margin": 8,
            "padding": 3,
            "width": 25,
          }
        }
        testID="eventListItemWeekDay"
      >
        <Text
          style={
            {
              "color": "#000000",
              "display": "flex",
              "flexDirection": "row",
              "fontSize": 12,
            }
          }
          testID="label"
        >
          S
        </Text>
      </View>
    </View>
  </View>
</RNGestureHandlerButton>
`;
